### ISSUE TITLE
kijiji_api.KijijiApiException: XSRF token not found in html text

### ISSUE THREAD

**Issue by iparamr** [ArthurG/Kijiji-Repost-Headless#180]
Not able to show ads anymore, I get the following error when I run the command:
`python kijiji_repost_headless [-u USERNAME] [-p PASSWORD] show`

```Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "kijiji_repost_headless/__main__.py", line 205, in <module>
    main()
  File "kijiji_repost_headless/__main__.py", line 51, in main
    args.function(args)
  File "kijiji_repost_headless/__main__.py", line 106, in show_ads
    api.login(args.username, args.password)
  File "kijiji_repost_headless/kijiji_api.py", line 111, in login
    "xsrfToken": get_xsrf_token(resp.text),
  File "kijiji_repost_headless/kijiji_api.py", line 84, in get_xsrf_token
    raise KijijiApiException("XSRF token not found in html text.", html)
kijiji_api.KijijiApiException: XSRF token not found in html text.
See kijijiapi_dump_20200530T191135.html in current directory for latest dumpfile.

**Comment by andr333y** [comment]
i got the same.
it was working like 7 days ago fine
i guess its captcha related

**Comment by iparamr** [comment]
Yes, it was working fine about a week ago. Not anymore 😩

**Comment by ArthurG** [comment]
Hmm, I'm looking at Kijiji and there still doesn't seem to be any Captcha on the login page. I think the change in #178 should still be valiid.

**Comment by iparamr** [comment]
@ArthurG are you able to login and show ads?

**Comment by ArthurG** [comment]
Nope.

**Comment by maltaisn** [comment]
Looks like the `https://www.kijiji.ca/t-login.html` page being scraped is some bogus obfuscated script: [kijijiapi_dump_20200531T194836.txt](https://github.com/ArthurG/Kijiji-Repost-Headless/files/4708704/kijijiapi_dump_20200531T194836.txt)

**Comment by scsikid** [comment]
Yea I still have the same issue. I pulled down the latest commit from master. Tried regenerating my ad, and yes I can log in to the kijiji UI with my credentials. 

```
Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "kijiji_repost_headless/__main__.py", line 200, in <module>
    main()
  File "kijiji_repost_headless/__main__.py", line 51, in main
    args.function(args)
  File "kijiji_repost_headless/__main__.py", line 87, in post_ad
    while not check_ad(args) and attempts < 5:
  File "kijiji_repost_headless/__main__.py", line 172, in check_ad
    api.login(args.username, args.password)
  File "kijiji_repost_headless/kijiji_api.py", line 111, in login
    "xsrfToken": get_xsrf_token(resp.text),
  File "kijiji_repost_headless/kijiji_api.py", line 84, in get_xsrf_token
    raise KijijiApiException("XSRF token not found in html text.", html)
kijiji_api.KijijiApiException: XSRF token not found in html text.
```

**Comment by RESP3CT88** [comment]
I have the same problem

**Comment by prgrm** [comment]
Is there any solution in sight?

**Comment by maltaisn** [comment]
Would it be possible to find out the API that the mobile app is using and use it? There's no captcha in the mobile app. And it definitely seems simpler than the scraping route.

**Comment by ArthurG** [comment]
I agree, using the mobile app API sounds like a good idea.

**Comment by halcyonts** [comment]
@rybodiddly has a working solution. [Kijiji Reposter](https://github.com/rybodiddly/Kijiji-Reposter) Maybe something from this project can be used?

**Comment by ArthurG** [comment]
I decompiled that code and found out that it is using a mobile Kijiji API - mingle.kijiji.ca

Sounds like the Mobile API could work

**Comment by rybodiddly** [comment]
@ArthurG Nice ;) Byte code is pretty easy to reverse. I just uploaded the original source [Here](https://github.com/rybodiddly/Kijiji-Reposter/blob/master/kijijiapi.py) The decomp is close, but it mangles the if statements a little. Anyways, use what you need.

**Comment by Corgano** [comment]
Maybe it's user agent related? Browser uses one agent, the reposter uses a different one, so it gives the browser the xsrf token and doesn't give it to the reposted?

**Comment by rybodiddly** [comment]
@Corgano The api has change once again. Captcha has now become a mandatory component in the exchange.

```
[{
	"operationName": "loginUser",
	"variables": {
		"input": {
			"emailOrNickname": "YOUR_USERNAME",
			"password": "YOUR_PASSWORD",
			"rememberMe": false,
			"recaptchaResponse": "03AGdBq26Shv3Rg18dex0ei-rzRQIKg17uxgtfTF65HUnyge6jli7hdbgdtr5436GTAabpXpXWpiZGNILHsqSoW0JVJZ1NXRUX-G2yQ0Z_HByUV6uqBKQGVspfZ-vY2xda5B8KZ3WcqOS10SjndD1pAssdeTtmVXe4D5xGOrx9lSUkhDeVkJZuS6y1zTyUJF7HKkTE-KslXsn0dl5v07RXqWEFBhewICCKeQFm_FUWz5TeHbdGoBT7iqMiPobmCtVHrabTKos0vd-YyM2vkmb0njp92c6cRUhLRpACFGtbPg4CvZZXKmNqWvtWWY_dfa1mS7RlnVpyuTdU8yreOg7EcYVR5RqDw8rBxPB9KPsdRciCQcbd8Zm5bg4dCKvi6jdHimPjbtEztLdxn",
			"targetUrl": null,
			"fraudToken": "032679048394577c4303497645gtdbs3",
			"campaign": null,
			"xsrfToken": "1591544939607.6e2f926f12a31bc2d360d4d647hyfgt546hyhr553d6f0c709f9fb6fe69d41614",
			"hints": ["NEW_AJAX_LOGIN"]
		}
	},
	"query": "mutation loginUser($input: LoginUserInput!) {\n  loginUser(input: $input) {\n    userId\n    message\n    statusCode\n    redirectUrl\n    __typename\n  }\n}\n"
}]
```

**Comment by vowswithin** [comment]
so shes shot for now?

**Comment by rybodiddly** [comment]
@vowswithin It is until a recaptcha patch is developed. So this thread should probably be closed since the issue isn't actually an XSRF token problem.

**Comment by ArthurG** [comment]
I believe this is fixed in #181. Please open a new issue if you find it still occurs.
